File: app/src/main/java/com/example/calculator/CalculatorBroadcastReceiver.java
package com.example.calculator;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.widget.Toast;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;

public class CalculatorBroadcastReceiver extends BroadcastReceiver {
    
    @Override
    public void onReceive(Context context, Intent intent) {
        try {
            String action = intent.getAction();
            
            if (action != null) {
                String message = "";
                
                // 根据接收到的广播类型显示不同的消息
                switch (action) {
                    case "com.example.calculator.CUSTOM_ACTION":
                        // 获取自定义广播中的额外数据
                        String customMessage = intent.getStringExtra("message");
                        long timestamp = intent.getLongExtra("timestamp", System.currentTimeMillis());
                        
                        // 格式化时间戳
                        SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss", Locale.getDefault());
                        String timeString = sdf.format(new Date(timestamp));
                        
                        // 构建显示消息
                        if (customMessage != null && !customMessage.isEmpty()) {
                            message = customMessage + "\n接收时间: " + timeString;
                        } else {
                            message = "收到自定义广播消息\n接收时间: " + timeString;
                        }
                        break;
                        
                    case Intent.ACTION_BATTERY_LOW:
                        message = "电池电量低，请连接充电器";
                        break;
                        
                    case Intent.ACTION_POWER_CONNECTED:
                        message = "充电器已连接";
                        break;
                        
                    default:
                        message = "收到未知广播: " + action;
                        break;
                }
                
                // 显示消息
                Toast.makeText(context, message, Toast.LENGTH_LONG).show();
            }
        } catch (Exception e) {
            e.printStackTrace();
            Toast.makeText(context, "处理广播时出错: " + e.getMessage(), Toast.LENGTH_LONG).show();
        }
    }
} 

File: app/src/main/java/com/example/calculator/CalculatorBroadcastReceiver.kt
package com.example.calculator

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.widget.Toast
import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale

/**
 * 计算器应用的广播接收器
 * 用于接收系统广播和自定义广播
 */
class CalculatorBroadcastReceiver : BroadcastReceiver() {
    
    override fun onReceive(context: Context, intent: Intent) {
        when (intent.action) {
            // 接收自定义广播
            "com.example.calculator.CUSTOM_ACTION" -> {
                val message = intent.getStringExtra("message") ?: "无消息内容"
                val timestamp = intent.getLongExtra("timestamp", 0L)
                val time = SimpleDateFormat("HH:mm:ss", Locale.getDefault()).format(Date(timestamp))
                
                Toast.makeText(
                    context, 
                    "收到广播: $message\n时间: $time", 
                    Toast.LENGTH_LONG
                ).show()
            }
            
            // 接收电池电量低广播
            Intent.ACTION_BATTERY_LOW -> {
                Toast.makeText(
                    context,
                    "电池电量低，请连接充电器",
                    Toast.LENGTH_LONG
                ).show()
            }
            
            // 接收充电器连接广播
            Intent.ACTION_POWER_CONNECTED -> {
                Toast.makeText(
                    context,
                    "充电器已连接",
                    Toast.LENGTH_SHORT
                ).show()
            }
        }
    }
} 